addons/content-index/src/main/java/org/commonjava/indy/content/index/IndexingContentManagerDecorator.java

```diff
@@ -228,25 +228,28 @@ else if ( indexCfg.isAuthoritativeIndex() && store.isAuthoritativeIndex() )
 
             if ( specialPathInfo == null || !specialPathInfo.isMergable() )
             {
-                transfer = getIndexedMemberTransfer( (Group) store, path, TransferOperation.DOWNLOAD,
-                                                     ( member ) -> {
-                                                         try
-                                                         {
-                                                             return delegate.retrieve( member, path );
-                                                         }
-                                                         catch ( IndyWorkflowException e )
-                                                         {
-                                                             logger.error( String.format(
-                                                                     "Failed to retrieve() for member path: %s:%s. Reason: %s",
-                                                                     member.getKey(), path, e.getMessage() ), e );
-                                                         }
-
-                                                         return null;
-                                                     } );
+                for ( StoreKey memberKey : ( (Group) store ).getConstituents() )
+                {
+                    try
+                    {
+                        ArtifactStore member = storeDataManager.getArtifactStore( memberKey );
+                        if ( member != null )
+                        {
+                            transfer = retrieve( member, path, eventMetadata );
```

Comment:
Somewhere after this line, if the transfer exists it should be indexed with the group that triggered the recursion too. Maybe that's present, and I missed it.

This way, when the recursion unwraps, each triggering group gets an index entry. It's a little slower in terms of setting up the index entries, but should be a much more resilient way to ensure different group configurations can take advantage of indexed paths.